ggml/src/ggml.c

```diff
@@ -6964,6 +6964,87 @@ void ggml_graph_print(const struct ggml_cgraph * cgraph) {
     GGML_LOG_INFO("========================================\n");
 }
 
+static int ggml_find_tensor_node_list(const struct ggml_cgraph * cgraph,
+                                      const int                *  idxs,
+                                      int                        count,
+                                      const struct ggml_tensor * tensor) {
+    if (idxs == NULL || cgraph == NULL) {
+        return -1;
+    }
+
+    for (int i = 0; i < count; ++i) {
+        const int node_idx = idxs[i];
+
+        if (node_idx >= cgraph->n_nodes) {
+            return -1;
+        }
+        if (cgraph->nodes[node_idx] == tensor) {
+            return i;
+        }
+    }
+    return -1;
+}
+
+bool ggml_can_fuse_subgraph_ext(const struct ggml_cgraph * cgraph,
+                                const int *                node_idxs,
+                                int                        count,
+                                const enum ggml_op *       ops,
+                                const int *                outputs,
+                                int                        num_outputs) {
+    GGML_ASSERT(count < 32 && outputs && num_outputs > 0);
+    int interior_nodes_count = 0;
+    int interior_nodes[32];
+
+    for (int i = 0; i < count; ++i) {
+        if (node_idxs[i] >= cgraph->n_nodes || cgraph->nodes[node_idxs[i]]->op != ops[i]) {
+            return false;
+        }
+
+        const struct ggml_tensor * node = cgraph->nodes[node_idxs[i]];
+
+        if (node->flags & GGML_TENSOR_FLAG_OUTPUT) {
```

Comment:
Suggest moving this check after the `ggml_find_tensor_node_list(cgraph, outputs, num_outputs, node) != -1` check. It's OK for the output nodes to have the output flag, since they won't be elided.